File: volleywrapper/src/androidTest/java/com/hipsheep/volleywrapper/network/request/GetRequest.java
package com.hipsheep.volleywrapper.java.network.request;

import com.hipsheep.volleywrapper.java.network.Params;
import com.hipsheep.volleywrapper.java.network.model.Post;
import com.hipsheep.volleywrapper.utils.GsonUtils;

/**
 * Test GET request that will be sent to the server.
 *
 * @author Franco Sabadini (fsabadi@gmail.com)
 */
public class GetRequest extends GsonRequest<Post[]> {

	public GetRequest(String url, Integer userId) {
		super(Method.GET, url, Post[].class, GsonUtils.getGson());

		// Set query params
		addQueryParam(Params.KEY_USER_ID, userId);
	}

}

File: volleywrapper/src/androidTest/java/com/hipsheep/volleywrapper/network/request/GetRequest.kt
package com.hipsheep.volleywrapper.network.request

import com.hipsheep.volleywrapper.network.Param
import com.hipsheep.volleywrapper.network.model.Post
import com.hipsheep.volleywrapper.utils.GsonUtils

/**
 * Test GET request that will be sent to the server.
 *
 * @author Franco Sabadini (fsabadi@gmail.com)
 */
class GetRequest(url: String, userId: Int) : GsonRequest<Array<Post>>(Method.GET, url, GsonUtils.gson) {

    init {
        addQueryParam(Param.Key.USER_ID, userId)
    }

}